scipy/stats/_resampling.py

```diff
@@ -314,6 +315,12 @@ def bootstrap(data, statistic, *, n_resamples=9999, batch=None,
             The bootstrap standard error, that is, the sample standard
             deviation of the bootstrap distribution
 
+    Warns
+    -----
+    `~scipy.stats.DegenerateDataWarning`
```

Comment:
Why doesn't the normal referencing work? Just `DegenerateDataWarning` should work, if not it's indicating some problem that should be solved.

Reply:
Probably `.. currentmodule:` being set incorrectly at the top of a `.rst`  file or some such thing.

Reply:
I was following an example written for result objects that weren't public, but maybe normal referencing will work now that these are public. I'll try it.

Reply:
The links are no longer rendering. Is there something easy to fix, or shall I revert?

Reply:
Let me have a look to see if I can figure this one out.

Reply:
I think the issue is the funky thing going on at https://github.com/scipy/scipy/blob/main/doc/source/reference/stats._result_classes.rst. I'll have a go at fixing that.

Reply:
Argh no, that isn't it. This does look weird. If only building the docs was a little faster ....

Reply:
Nope, I can't find it so quickly. Please revert @mdhaber

Reply:
Reverted.